app/src/main/java/org/oppia/android/app/drawer/ExitProfileDialogFragment.kt

```diff
@@ -38,33 +42,47 @@ class ExitProfileDialogFragment : DialogFragment() {
     val args =
       checkNotNull(arguments) { "Expected arguments to be pass to ExitProfileDialogFragment" }
 
-    val isFromNavigationDrawer = args.getBoolean(
-      BOOL_IS_FROM_NAVIGATION_DRAWER_EXTRA_KEY,
-      false
+    val exitProfileDialogArguments = args.getProto(
+      EXIT_PROFILE_DIALOG_ARGUMENTS_PROTO,
+      ExitProfileDialogArguments.getDefaultInstance()
     )
 
-    if (isFromNavigationDrawer) {
+    val restoreLastCheckedItem = when (exitProfileDialogArguments.highlightItemCase) {
+      ExitProfileDialogArguments.HighlightItemCase.HIGHLIGHT_NONE -> false
+      else -> true
```

Comment:
I think you might want to invert this (e.g. only set it true for the cases where restoration should happen) since there are auto-generated constants being included in the 'else' (such as unspecified).

Reply:
Makes sense, done.